src/main/java/frc/robot/subsystems/snapback/hood/Hood.java

```diff
@@ -1,3 +1,96 @@
 package frc.robot.subsystems.snapback.hood;
 
-public class Hood {}
+import static edu.wpi.first.units.Units.*;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine.Direction;
+import org.littletonrobotics.junction.Logger;
+
+public class Hood extends SubsystemBase {
+  private final HoodIO io;
+  private final HoodIOInputsAutoLogged inputs;
+
+  private SysIdRoutine characterizationRoutine;
+
+  public Hood(HoodIO io) {
+    inputs = new HoodIOInputsAutoLogged();
+    this.io = io;
+
+    characterizationRoutine =
+        new SysIdRoutine(
+            new SysIdRoutine.Config(
+                Volts.of(0.5).per(Seconds),
+                Volts.of(3.5),
+                Seconds.of(10),
+                (state) -> Logger.recordOutput("Arm/sysIDState", state.toString())),
+            new SysIdRoutine.Mechanism((volts) -> io.setVoltage(volts.in(Volts)), null, this));
+  }
+
+  public void periodic() {
+    io.updateInputs(inputs);
+    Logger.processInputs("Hood", inputs);
+  }
+
+  public Command setVoltage(double volts) {
+    return runOnce(() -> io.setVoltage(volts));
+  }
+
```

Comment:
We need a method to set the hood to the speaker angle. Use the robot state calculation for this.